pkg/machine/convert.go

```diff
@@ -49,6 +49,10 @@ func GetAPIV1OperatingSystemSpec(machineSpec clusterv1alpha1.MachineSpec) (*apiv
 		return nil, fmt.Errorf("failed to get machine providerConfig: %v", err)
 	}
 
+	if decodedProviderSpec.OperatingSystemSpec.Raw == nil {
+		return &apiv1.OperatingSystemSpec{}, nil
+	}
```

Comment:
Down the callstack the empty OperatingSystemSpec{} will cause the "no operating system set" error:

https://github.com/kubermatic/kubermatic/blob/e34f174049c563225eb6fbafc66c5c74ca6426b7/pkg/machine/util.go#L86-L96

Reply:
Well, I am not sure why this is not using the OS filed directly! At any case I am actually still testing it :)